File: ransom-note/code.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        magazineCount = {}

        for value in magazine:
            if value not in magazineCount:
                magazineCount[value] = 1
            else:
                magazineCount[value] += 1

        for value in ransomNote:
            if value not in magazineCount or magazineCount[value] <= 0 :
                return False
            else:
                magazineCount[value] -= 1
        return True

File: ransom-note/code.ts
function canConstructTs(ransomNote: string, magazine: string): boolean {
    let magazineMap: Map<string, number> = new Map()
    for(const char of magazine){
        if(magazineMap.get(char) === undefined){
            magazineMap.set(char, 1)
        }else{
            magazineMap.set(char, magazineMap.get(char)! + 1)
        }
    }

    for (const char of ransomNote){
        if(magazineMap.get(char) === 0 || magazineMap.get(char) === undefined){
            return false
        }
        magazineMap.set(char, magazineMap.get(char)! - 1)
    }
    return true
};